### PR TITLE
Dom storage decorators

### PR DESCRIPTION
**Pull request by bayes343**
Adds dom storage decorators following the same api as the state and appstate decorators.

### CODE REVIEW

File: src/core/decorators/storage.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    ) {
+      return JSON.parse(rawValue.Value);
+    } else {
+      return rawValue.Value;
+    }
+  };
+
+  const setter = function (newValue) {
+    const valueToStore = typeof newValue === 'object' ?
+      JSON.stringify(newValue) : newValue;
```
Comment (sdapalsch) [inline src/core/decorators/storage.ts:26]:
```suggestion
    const valueToStore = typeof newValue !== 'string' ?
      JSON.stringify(newValue) : newValue;
```

File: src/core/decorators/storage.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        rawValue.Value.startsWith('{') ||
+        rawValue.Value.startsWith('[') ||
+        !isNaN(parseFloat(rawValue.Value))
+      )
+    ) {
+      return JSON.parse(rawValue.Value);
+    } else {
+      return rawValue.Value;
+    }
+  };
```
Comment (sdapalsch) [inline src/core/decorators/storage.ts:22]:
```suggestion
    const result = storage.GetValue(key);
    if (
      result.IsSuccess &&
      typeof result.Value === 'string' &&
      (
        result.Value.startsWith('{') ||
        result.Value.startsWith('[') ||
        !isNaN(parseFloat(result.Value))
      )
    ) {
      return JSON.parse(result.Value);
    } else {
      return result.Value;
    }
  };
```

**Review by sdapalsch — APPROVED** [review]
Consider checking if the value is a string in the setter and storing that information along the value somehow. Then in the getter you can leave it a string if it was originally a string and only parse it if it was something else.

**Comment by bayes343** [comment]
Thanks for the suggestion @sdapalsch!

**Pull request merged**